source/mqtt_client_connection.c

```diff
@@ -584,6 +747,9 @@ static void s_on_connect_call(napi_env env, napi_value on_connect, void *context
     if (args->return_code || args->error_code) {
         /* Failed to create a connection, none of the callbacks will be invoked again */
         s_mqtt_client_connection_release_threadsafe_function(binding);
+        if (binding->first_successfull_connection == true) {
```

Comment:
Looking into the code, it seems the `first_successfull_connection` will never be true here.Is it necessary then?

Other than that, PR looks good to me.